crates/ide/src/folding_ranges.rs

```diff
@@ -57,6 +58,31 @@ pub(crate) fn folding_ranges(file: &SourceFile) -> Vec<Fold> {
                 NodeOrToken::Token(token) => token.text().contains('\n'),
             };
             if is_multiline {
+                // for the arg list, we need to special handle
+                if matches!(element.kind(), ARG_LIST | PARAM_LIST) {
+                    if let NodeOrToken::Node(node) = &element {
+                        if let Some(fn_node) = node.parent().and_then(ast::Fn::cast) {
+                            if let Some(body) = fn_node.body() {
+                                // just add a big fold combine the params and body
+                                res.push(Fold {
+                                    range: TextRange::new(
+                                        node.text_range().start(),
+                                        body.syntax().text_range().end(),
+                                    ),
+                                    kind: FoldKind::ArgList,
```

Comment:
This doesn't seem particularly more `ArgList` than body, perhaps add s new kind?